File: rust/src/base/models/square.rs
use crate::base::commons::Dimensional;

pub struct Square {
    pub side: f64,
}

impl Square {
    pub fn new(side: f64) -> Self {
        let dim_one: f64 = if side < 0.0 { 1.0 } else { side };
        Self { side: dim_one }
    }
}

impl Dimensional for Square {
    fn area(&self) -> f64 {
        self.side.powi(2)
    }

    fn perimeter(&self) -> f64 {
        self.side * 4.0
    }
}

File: php/src/Base/Models/Square.php
<?php

namespace Shapes\Base\Models;

use Shapes\Base\Commons\Dimensional;

final class Square implements Dimensional {
    private float $side;

    public function __construct(float $side) {
        $dim_one = $side > 0 ? $side : 0;
        $this->side = $dim_one;
    }

    #[\Override]
    public function area(): float {
        return $this->side ** 2;
    }

    #[\Override]
    public function perimeter(): float {
        return 4.0 * $this->side;
    }
}
